### ISSUE TITLE
False positive "consider-using-f-string" for DataFrame formatting

### ISSUE THREAD

**Issue by DiegoF90** [PyCQA/pylint#7262]
### Bug description

```
import pandas as pd

df = pd.DataFrame([[1]], columns=['a'])
df['a'].map('{0:+.2f}'.format).astype(str)  # pylint: disable=consider-using-f-string

```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:4:12: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)

------------------------------------------------------------------
Your code has been rated at 6.67/10
```

### Expected behavior

No warning issued

### Pylint version

```shell
pylint 2.14.5
```

### Additional dependencies

pandas==1.1.5

**Comment by Pierre-Sassoulas** [comment]
I did not test this and I'm a little rusty with pandas, but ``str.format`` is very inefficient which make me think that this might not be a false positive. Would an alternative lambda syntax (``df['a'].map(lambda x: f'{+:+.2f}'')``) or list comprehension syntax (``df['a'] = [f'{x:+.2f}' for x in df['a']]``)  work and be better performance wise ?

**Comment by DiegoF90** [comment]
Thanks @Pierre-Sassoulas, list comprehension is indeed about twice as fast.
